x/tokenomics/keeper/update_relay_mining_difficulty.go

```diff
@@ -156,6 +159,25 @@ func computeEma(alpha float64, prevEma, currValue uint64) uint64 {
 	return uint64(alpha*float64(currValue) + (1-alpha)*float64(prevEma))
 }
 
+// RelayMiningTargetHashToDifficulty returns the relay mining difficulty based on the hash.
+// This currently implies the number of leading zero bits but may be changed in the future.
+// TODO_MAINNET: Determine if we should launch with a more adaptive difficulty or stick
+// to leading zeroes.
+func RelayMiningTargetHashToDifficulty(targetHash []byte) int {
```

Comment:
@red-0ne 

1. See how `LeadingZeroBitsToTargetDifficultyHash` has some unit tests for this
2. PTAL at @bryanchriswhite's PRs and identify if there's an opportunity to consolidate some helpers.
3. Looking at Bryan's more recent PR, do we even need the loop?